Copy the set_util() method into init(), add helper method for post type

kienstra opened the pull request:
* To follow the pattern of the `Plugin::init()` method, this moves the contents of `->set_util()` into that method, and deletes `->set_util()`
* Also, it adds a helper method `Plugin::get_post_type_slug()`
* This looks to remove the need to add `$block_post` as a property of Plugin
* This prevents `block_lab()->block_post` from essentially being in the global scope, so now people can't call something like `block_lab()->block_post->enqueue_scripts()` or `->add_caps()`

kienstra committed:
Copy the set_util() method into init(), add helper method for post type

To follow the pattern of the init() method,
this moves the contents of set_util() into
that method, and deletes the method.
Also, it adds a helper method
get_post_type_slug() to Plugin.
This might remove the need to add $block_post
as a property of Plugin.

kienstra committed:
Replace another reference to ->block_post->slug with helper method

I missed this in the previous commit,
so replace it with get_post_type_slug().

lukecarbis commented on php/class-plugin-abstract.php:
```diff
@@ -53,6 +53,14 @@ abstract class Plugin_Abstract implements Plugin_Interface {
+	 * @since 1.3.5
+	 * @var string
+	 */
+	protected $post_type_slug = 'block_lab';
```
Do you think this belongs in the plugin abstract? I imagine it would go in the `Plugin` class, not the abstract version of it (which to me, should be portable to different plugins).

kienstra replied:
Ah, good point. Abstract should apply to any plugin.

kienstra replied:
Good idea, https://github.com/getblocklab/block-lab/pull/384/commits/4614c78849587a6a2dc0b4ab04e287691212979b moves this to `Plugin`.

lukecarbis commented on php/class-plugin-abstract.php:
```diff
@@ -182,6 +183,15 @@ public function get_slug() {
+	 * @return string The slug.
+	 */
+	public function get_post_type_slug() {
+		return $this->post_type_slug;
```
This could be moved to the `Util` class?

Inside `Util`, it could just do this:
```
return $this->plugin->post_type_slug;
```

OR we could keep the post type slug in `Block_Post`, and do this instead:

```
return $this->plugin->block_post->slug;
```

The `plugin` property is set automatically when registering any component.

kienstra replied:
Great idea, this is now moved to `Util` in https://github.com/getblocklab/block-lab/pull/384/commits/4614c78849587a6a2dc0b4ab04e287691212979b

lukecarbis commented on php/class-plugin.php:
```diff
@@ -13,38 +13,30 @@
+	 * @var Util
 	 */
-	public $block_post;
+	protected $util;
```
I like this being protected.

kienstra replied:
Thanks!

lukecarbis commented on php/class-plugin.php:
```diff
@@ -13,38 +13,30 @@
 
-		$this->loader = new Blocks\Loader();
-		$this->register_component( $this->loader );
+		$this->register_component( new Post_Types\Block_Post() );
```
Nice work removing the `block_post` and `loader` properties.

kienstra replied:
Thanks 😄

kienstra committed:
Move get_post_type_slug() to from Plugin_Abstract Util

As Luke mentioned, Plugin_Abstract isn't the
best place for this,
as it should apply to any plugin.
So move it to Utils.

kienstra committed:
Fix failed unit tests by making a property public and changing setUp()

The setUp method should actually just
call ->set_plugin(),
instead of register_component().

lukecarbis commented on php/class-plugin.php:
```diff
@@ -13,38 +13,38 @@
+	 * @var string
 	 */
-	public $loader;
+	public $post_type_slug = 'block_lab';
```
What do you think about just calling this `$slug` instead of `$post_type_slug` – purely for brevity?

kienstra replied:
Yeah, that would be a little simpler. The reasoning was to distinguish it from the `$slug` property of `Plugin_Abstract`. 

Though both of them end up being `'block_lab'`

lukecarbis replied:
Aah! Good point. Where is that used?

lukecarbis replied:
Might be better to just go with the `Plugin_Abstract->$slug` property. The post type can just use that too.

kienstra replied:
Though maybe `Util::get_post_type_slug()` could simply return `Plugin::$slug`, which also happens to be `'block_lab'`

kienstra replied:
Ah, my fault. `Plugin::$slug` is `'block-lab'`, while `Plugin::$post_type_slug` is `'block_lab'`

lukecarbis reviewed (approved):
Nice work!

kienstra commented:
Thanks, Luke!
